src/auth-service/utils/create-preference.js

```
@@ -54,25 +54,40 @@
+      update["$addToSet"] = update["$addToSet"] || {};
+      update["$addToSet"][field] = { $each: update[field] };
       delete update[field];
     }
   });
 
+  // Remove simple ObjectId fields from being processed by $addToSet
+  const singleObjectIdFields = ["user_id", "group_id"];
+  singleObjectIdFields.forEach((field) => {
+    if (update[field]) {
+      // Keep the original value for simple ObjectId fields
+      update[field] = update[field];
```

Biome (1.9.4)

[error] field is assigned to itself.

This is where is assigned.


(lint/correctness/noSelfAssign)